### PR TITLE
Payments 2: Implements component

### PR DESCRIPTION
**Pull request by igorp1**
__What__

This PR introduces the Duffel payments component to our library. It comes with some breaking changes for users consuming the library through our CDN. 

- Two new examples have been added 
- The react example now also includes the payments component
- The `react-build` now exports the `DuffelPayments` component as well as the custom element helpers (`onDuffelPaymentsFailedPayment`, `onDuffelPaymentsSuccessfulPayment`, `renderDuffelPaymentsCustomElement`)  
- The `cdn-build` has been changed to support different components. Notice that examples will now import a specific component instead of a generic index file. `<script src="http://localhost:8000/index.js"></script>` ⇢ `<script src="http://localhost:8000/duffel-ancillaries.js"></script>`. Documentation update will follow. 

Here's a one of the new examples running to demonstrate importing the component with the `script` tag:

https://github.com/duffelhq/duffel-components/assets/2934495/10372311-1198-4e40-80f1-60a6c6763cf2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,4 +57,8 @@ The ancillaries component allows your customers to add ancillaries to their orde
 
 - [Find live demo on codesandbox.io&nbsp;&nbsp;↗](https://codesandbox.io/s/duffel-ancillaries-example-1nxuu7)
 
+### Payments component
+
+Tha payments component provides a [PCI-compliant](https://help.duffel.com/hc/en-gb/articles/4409049543058) way for you to collect card payments for online bookings from your customers. To learnh more about how to work with Duffel payments please visit the [Collecting customer card payments guide](https://duffel.com/docs/guides/collecting-customer-card-payments).
```
Comment (shaundon) [inline README.md:62]:
```suggestion
Tha payments component provides a [PCI-compliant](https://help.duffel.com/hc/en-gb/articles/4409049543058) way for you to collect card payments for online bookings from your customers. To learn more about how to work with Duffel payments please visit the [Collecting customer card payments guide](https://duffel.com/docs/guides/collecting-customer-card-payments).
```

File: src/examples/payments-custom-element/index.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    const duffelpaymentsElement = document.querySelector("duffel-payments");
+
+    // 3. Render the component with the required data, you can safely call this function as many times as you want. E.g. when your passenger data changes.
+    duffelpaymentsElement.render({
+      paymentIntentClientToken:
+        "eyJjbGllbnRfc2VjcmV0IjoicGlfM0psczlVQWcySmhFeTh2WTBSTm1MU0JkX3NlY3JldF9QUW9yZXNuU3laeWJadGRiejZwNzBCbUdPIiwicHVibGlzaGFibGVfa2V5IjoicGtfdGVzdF9EQUJLY0E2Vzh6OTc0cTdPSWY0YmJ2MVQwMEpwRmMyOUpWIn0=",
+      debug: true,
+    });
+
+    // 4. Listen to 'onPayloadReady' event on the component. `event.detail.data` contains the payload you need to send to Duffel's API to create an order.
```
Comment (shaundon) [inline src/examples/payments-custom-element/index.html:33]: Is this comment still correct? It looks like it takes no arguments

**Review by shaundon — CHANGES_REQUESTED** [review]
This is great! I have a couple of tiny comments, but otherwise looks great. I think this is a really strong vindication of the approach we've adopted, evidenced by how the method for consumers to use this component fits in really nicely and is consistent with the ancillaries component

File: README.md (hunk 1 above)
Comment (igorp1) [reply README.md:62]: Ah good catch! And the `Tha` in the beginning too, just fixed it!

File: src/examples/payments-custom-element/index.html (hunk 2 above)
Comment (igorp1) [reply src/examples/payments-custom-element/index.html]: Def not, good spot, just fixed it!

**Review by shaundon — APPROVED** [review]

**Pull request merged**